scripts/check_ecosystem.py

```diff
@@ -45,11 +45,11 @@ async def clone(self: Self, checkout_dir: Path) -> AsyncIterator[Path]:
         """Shallow clone this repository to a temporary directory."""
         if checkout_dir.exists():
             logger.debug(f"Reusing {self.org}:{self.repo}")
-            yield Path(checkout_dir)
+            yield await self._get_commit(checkout_dir)
```

Comment:
So... because `Repository` is immutable (and I liked that it was immutable) we can't just get the commit SHA on clone and stash it on the `Repository` object. Instead, we yield it and it's attached to the `Diff` object for later display. It also turns out that the previously yielded `Path` was just an input argument so I removed that.

We probably ought to just have the ecosystem checks pinned to a specific SHA that's updated on a schedule by a bot and then we can skip the retrieval and just assign it to the `Repository` at creation time. That seems pretty low priority though.

Reply:
Also happy to hear alternative approaches if anyone has ideas! This seemed liked the simplest way to retain immutability but this was my first time looking at this script.

Reply:
We could [replace](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace) the value:
```python
self._replace(ref=await self._get_commit(checkout_dir))
```

But, otherwise I think this is fine.

Reply:
I'm not sure how we'd get the new `Repository` object back to the relevant owner without a bunch of changes.

Reply:
Ah yeah, didn't think about that (`_replace` returns a new object). I think this is fine then. We can iterate it later.